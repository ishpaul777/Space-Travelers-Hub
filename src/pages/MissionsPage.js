import Mission from '../Components/missions/mission';

const MissionsPage = () => (
  <div>
    <Mission />
  </div>
);

export default MissionsPage;
