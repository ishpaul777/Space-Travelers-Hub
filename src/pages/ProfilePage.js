import './Styles/ProfilePage.css';
import { useSelector } from 'react-redux';

const ProfilePage = () => {
  const state = useSelector((state) => state);
  let rocketsReserved = 0;
  let missionJoined = 0;
  return (
    <section className="profile--info--container flex">
      <div className="rockets--booked--info">
        <h2>My Missions</h2>
        <table>
          <tbody>
            {
              state.rockets.map((rocket) => {
                if (rocket.isReserved) {
                  rocketsReserved += 1;
                  return (
                    <tr key={rocket.id}>
                      <td>{rocket.name}</td>
                    </tr>
                  );
                }
                return null;
              })
            }
            {
              !rocketsReserved
              && (
                <tr>
                  <td>No Rockets Reserved</td>
                </tr>
              )
            }
          </tbody>
        </table>
      </div>
      <div className="missions--joined--info">
        <h2>My Rockets</h2>
        <table>
          <tbody>
            {
              state.missions.map((mission) => {
                if (mission.isjoined) {
                  missionJoined += 1;
                  return (
                    <tr key={mission.id}>
                      <td>{mission.name}</td>
                    </tr>
                  );
                }
                return null;
              })
            }
            {
              !missionJoined
              && (
                <tr>
                  <td>No Missions Joined</td>
                </tr>
              )
            }

          </tbody>
        </table>
      </div>
    </section>
  );
};

export default ProfilePage;
