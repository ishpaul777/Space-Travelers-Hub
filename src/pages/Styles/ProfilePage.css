.profile--info--container {
  margin-top: 80px;
  align-items: flex-start;
  justify-content: center;
  gap: 10vw;
  flex-wrap: wrap;
}

.profile--info--container table {
  margin-top: 20px;
  border-collapse: collapse;
  width: 30vw;
}

.profile--info--container td {
  border: 1px solid #868686;
  text-align: left;
  padding: 13px;
}
