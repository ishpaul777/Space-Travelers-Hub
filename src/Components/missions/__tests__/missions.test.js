import {
  fireEvent, render, screen,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import ProfilePage from '../../../pages/ProfilePage';
import missionReducer from '../../../redux/Missions/missions';
import rocketReducer from '../../../redux/Rockets/rockets';
import MissionsPage from '../../../pages/MissionsPage';

const initialState = {
  missions: [{
    id: 'Thaicom#123',
    name: 'Thaicom',
    description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
    isjoined: false,
  }],
};

const rootReducer = combineReducers({
  missions: missionReducer,
  rockets: rocketReducer,
});

const store = configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});

describe('Missions tests', () => {
  test('Mission Page should render', () => {
    const missionPage = render(
      <Provider store={store}>
        <BrowserRouter>
          <MissionsPage />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(missionPage).toMatchSnapshot();
  });

  test('Displays the Mission', () => {
    render(
      <Provider store={store}>
        <MissionsPage />
      </Provider>,
    );
    const missionName = screen.getByText('Thaicom');
    expect(missionName).toBeInTheDocument();
  });

  test('Joining a Mission and reflecting the joined mission in Profile Page', () => {
    render(
      <Provider store={store}>
        <MissionsPage />
        <ProfilePage />
      </Provider>,
    );
    expect(screen.getByText('No Missions Joined')).toBeInTheDocument();

    fireEvent.click(screen.getByText('Join Mission'));
    expect(screen.getAllByText('Thaicom').length).toEqual(2);
  });
});
