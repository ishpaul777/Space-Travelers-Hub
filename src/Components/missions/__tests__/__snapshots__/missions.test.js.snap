// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions tests Mission Page should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="missions--container flex"
        >
          <table>
            <thead>
              <tr>
                <th>
                  Mission
                </th>
                <th>
                  Description
                </th>
                <th
                  colspan="2"
                >
                  Status
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="mission--name"
                >
                  Thaicom
                </td>
                <td
                  class="mission--description"
                >
                  Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
                </td>
                <td>
                  <span
                    class="joined--status--false"
                  >
                    Not a member
                  </span>
                </td>
                <td>
                  <button
                    class="join--mission--button"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="missions--container flex"
      >
        <table>
          <thead>
            <tr>
              <th>
                Mission
              </th>
              <th>
                Description
              </th>
              <th
                colspan="2"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="mission--name"
              >
                Thaicom
              </td>
              <td
                class="mission--description"
              >
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
              </td>
              <td>
                <span
                  class="joined--status--false"
                >
                  Not a member
                </span>
              </td>
              <td>
                <button
                  class="join--mission--button"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
